File: 2025-10/2025-10-31/SpOoKy~CaSe.js
function spookify(boo) {
  const spooky = boo.replace(/[_-]/g, "~").split("");
  let everyOtherLetter = true;

  for (const index in spooky) {
    if (/[a-zA-Z]/.exec(spooky[index])) {

      spooky[index] = everyOtherLetter ? spooky[index].toUpperCase() : spooky[index].toLowerCase();
      everyOtherLetter = !everyOtherLetter;
    }
  }

  return spooky.join("");
}

console.log(spookify("hello_world"));
console.log(spookify("Spooky_Case"));
console.log(spookify("TRICK-or-TREAT"));
console.log(spookify("c_a-n_d-y_-b-o_w_l"));
console.log(spookify("thE_hAUntEd-hOUsE-Is-fUll_Of_ghOsts"));

File: 2025-10/2025-10-31/SpOoKy~CaSe.py
import re

def spookify(boo):
    spooky = list(re.sub("[_-]", "~", boo))
    everyOtherLetter = True

    for index, character in enumerate(spooky):
        if re.match("[a-zA-Z]", character):
            
            spooky[index] = character.upper() if everyOtherLetter else character.lower()
            everyOtherLetter = not everyOtherLetter

    return "".join(spooky)

print(spookify("hello_world"))
print(spookify("Spooky_Case"))
print(spookify("TRICK-or-TREAT"))
print(spookify("c_a-n_d-y_-b-o_w_l"))
print(spookify("thE_hAUntEd-hOUsE-Is-fUll_Of_ghOsts"))